.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,67 @@
+name: zzz [DO NOT RUN] Automated unit tests
+
+on:
+  workflow_dispatch: # TODO: remove this after testing
+  pull_request:
+    branches:
+      - dev
+      - main
+  push:
+    branches:
+      - dev
+      - main
+
+jobs:
+  test:
+    name: Run Unit Tests
+    runs-on: macos-15
+    #if: github.repository_owner == 'nightscout' # TODO: remove comment after testing
+
+    steps:
+      - name: Select Xcode version
+        run: sudo xcode-select -s /Applications/Xcode_16.3.app/Contents/Developer
+
```

Comment:
I would suggest to introduce caching here, to improve the runtime of the overall workflow.
Example:
```yaml
- name: Cache DerivedData and .build
        uses: actions/cache@v4
        with:
          path: |
            ~/Library/Developer/Xcode/DerivedData
            .build
          key: ${{ runner.os }}-trio-${{ hashFiles('**/*.swift') }}
          restore-keys: |
            ${{ runner.os }}-trio-
```